src/xarray_regrid/utils.py

```diff
@@ -89,23 +90,32 @@ def create_regridding_dataset(
     )
 
 
-def to_intervalindex(coords: np.ndarray, resolution: float) -> pd.IntervalIndex:
-    """Convert a list of (regularly spaced) 1-d coordinates to pandas IntervalIndex.
+def to_intervalindex(coords: np.ndarray) -> pd.IntervalIndex:
```

Comment:
This takes care of #6 and #25, and I think this makes reasonable assumptions but let me know what you think